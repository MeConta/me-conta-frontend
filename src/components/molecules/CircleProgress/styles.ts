import styled from 'styled-components'

export const CircleContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-width: 9rem;
  max-width: 11rem;
`

interface CircleLineProp {
  active?: boolean
  displayLine?: boolean
}

const transitionTime = 'all 1s'

export const Circle = styled.div<CircleLineProp>`
  display: flex;
  justify-content: center;
  width: 7rem;
  height: 7rem;
  background-color: ${(p) =>
    p.active ? p.theme.colors.cornflowerBlue : p.theme.colors.transparent};
  border-radius: 50%;
  transition: ${transitionTime};
`

export const CircleLine = styled.div<CircleLineProp>`
  display: flex;
  width: 8rem;
  height: 8rem;
  background-color: ${(p) => p.theme.colors.transparent};
  border-radius: 50%;
  border: ${(p) => (p.displayLine ? '1px solid ' : '')};
  border-color: ${(p) =>
    p.active ? p.theme.colors.cornflowerBlue : p.theme.colors.spanishGray};
  justify-content: center;
  align-items: center;
  transition: ${transitionTime};
`

export const CircleSubtitle = styled.p<CircleLineProp>`
  width: inherit;
  color: ${(p) =>
    p.active ? p.theme.colors.cornflowerBlue : p.theme.colors.spanishGray};
  font-weight: ${(p) => p.active && p.displayLine && p.theme.font.bold};
  font-size: ${(p) => p.theme.font.sizes['desk-large']};
  text-align: center;
  transition: ${transitionTime};
`

interface IconProp {
  active?: boolean
  paddingLeft?: string
}

export const Icon = styled.div<IconProp>`
  display: flex;
  order: 1;
  padding-left: ${(p) => p.paddingLeft || p.theme.zero};

  & > svg {
    width: 4rem;
    height: 100%;
    color: ${(p) =>
      p.active ? p.theme.colors.white : p.theme.colors.spanishGray};
    transition: ${transitionTime};
  }
`
