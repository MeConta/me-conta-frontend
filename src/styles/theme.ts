// eslint-disable-next-line import/no-anonymous-default-export
export default {
  grid: {
    container: '130rem',
    gutter: '3.2rem'
  },
  border: {
    'btn-round-radius': '3.4rem',
    'btn-square-radius': '0.7rem'
  },
  font: {
    family:
      "Mulish, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      'desk-xsmall': '1rem',
      'desk-small': '1.1rem',
      'desk-medium': '1.2rem',
      'desk-large': '1.5rem',
      'desk-xlarge': '2.2rem',
      'desk-xxlarge': '3rem',
      'mob-xsmall': '0.8rem',
      'mob-small': '1rem',
      'mob-medium': '1.2rem',
      'mob-large': '1.5rem',
      'mob-xlarge': '1.8rem'
    }
  },
  colors: {
    maroonFlush: '#B2244D',
    ceriseRed: '#DE3163',
    cobalt: '#0545AB',
    cornflowerBlue: '#458FF6',
    malibu: '#67C3F3',
    mineShaft: '#333333',
    lightGray: '#848A8C',
    gray: '#828282',
    powderAsh: '#BBC7C3',
    white: '#FFFFFF',
    black: '#000000'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
    xxxlarge: '6.4rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
}
