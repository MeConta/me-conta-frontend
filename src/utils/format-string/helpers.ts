// Rever tipagem
export const formatPhoneNumber = (value: string): any =>
  value
    .replace(/\D/g, '')
    .replace(/(\d{2})(\d)/, '($1) $2')
    .replace(/(\s\d{1})/, '$1 ')
    .replace(/(\d{4})(\d)/, '$1-$2')
    .replace(/(-\d{4})(\d+?)$/, '$1')

export const removePhoneMask = (value: string) => value.replace(/\D/g, '')
