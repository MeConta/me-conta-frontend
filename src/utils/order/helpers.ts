import { VolunteerAvailableSlot } from '../../services/volunteers-service/volunteer-service'

function convertSlotsIntoString(dateSlot: VolunteerAvailableSlot) {
  return new Date(dateSlot.inicio).toLocaleTimeString([], {
    hour: '2-digit',
    minute: '2-digit',
    hour12: false
  })
}

export const sortSlotsByTime = (dateSlot: VolunteerAvailableSlot[]) => {
  return dateSlot.sort((a, b) => {
    const x = convertSlotsIntoString(a)
    const y = convertSlotsIntoString(b)
    return x == y ? 0 : x > y ? 1 : -1
  })
}
