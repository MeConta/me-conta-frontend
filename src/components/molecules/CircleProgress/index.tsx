import * as S from './styles'

interface CircleProp {
  active?: boolean
  displayLine?: boolean
  icon?: any
  paddingLeft?: string
  children?: string
}

export default function CircleProgress({
  active,
  displayLine,
  icon,
  paddingLeft,
  children
}: CircleProp) {
  return (
    <S.CircleContainer data-testid="circleContainer">
      <S.CircleLine
        active={active}
        displayLine={displayLine}
        data-testid="circleLine"
      >
        <S.Circle active={active} data-testid="circle">
          <S.Icon
            active={active}
            paddingLeft={paddingLeft}
            data-testid="circleIcon"
          >
            {icon}
          </S.Icon>
        </S.Circle>
      </S.CircleLine>
      <S.CircleSubtitle
        active={active}
        displayLine={displayLine}
        data-testid="circleSubtitle"
      >
        {children}
      </S.CircleSubtitle>
    </S.CircleContainer>
  )
}
