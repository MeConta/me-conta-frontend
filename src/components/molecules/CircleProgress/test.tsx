import { render, screen } from 'utils/tests/helpers'
import theme from '../../../styles/theme'
import { App } from '@styled-icons/bootstrap'

import CircleProgress from '.'

describe('<CircleProgress/>', () => {
  it('should render the component on screen', () => {
    render(<CircleProgress></CircleProgress>)

    expect(screen.getByTestId('circleContainer')).toBeInTheDocument()
    expect(screen.getByTestId('circleSubtitle')).toBeInTheDocument()
    expect(screen.getByTestId('circleLine')).toBeInTheDocument()
    expect(screen.getByTestId('circleIcon')).toBeInTheDocument()
    expect(screen.getByTestId('circle')).toBeInTheDocument()

    expect(screen.getByTestId('circleContainer')).toMatchSnapshot()
  })

  it('circle container should contain all elements', () => {
    render(<CircleProgress></CircleProgress>)

    const circleContainer = screen.getByTestId('circleContainer')
    expect(circleContainer).toBeInTheDocument()
    expect(circleContainer).toContainElement(screen.getByTestId('circle'))
    expect(circleContainer).toContainElement(screen.getByTestId('circleIcon'))
    expect(circleContainer).toContainElement(screen.getByTestId('circleLine'))
    expect(circleContainer).toContainElement(
      screen.getByTestId('circleSubtitle')
    )
  })

  it('should render circle with cornflowerblue color when active is true', () => {
    render(<CircleProgress active={true} displayLine={false}></CircleProgress>)

    expect(screen.getByTestId('circle')).toHaveStyle({
      'background-color': theme.colors.cornflowerBlue
    })
    expect(screen.getByTestId('circleSubtitle')).toHaveStyle({
      color: theme.colors.cornflowerBlue
    })
  })

  it('should render circle line with cornflowerblue color and subtitle bold when active and displayLine are true', () => {
    render(<CircleProgress active={true} displayLine={true}></CircleProgress>)

    expect(screen.getByTestId('circleLine')).toHaveStyle({
      border: '1px solid',
      'border-color': theme.colors.cornflowerBlue
    })
    expect(screen.getByTestId('circleSubtitle')).toHaveStyle({
      'font-weight': theme.font.bold,
      color: theme.colors.cornflowerBlue
    })
  })

  it('should render circle with spanishGray color when active is false', () => {
    render(<CircleProgress active={false} displayLine={false}></CircleProgress>)

    expect(screen.getByTestId('circleLine')).toHaveStyle({
      'border-color': theme.colors.spanishGray
    })
    expect(screen.getByTestId('circle')).toHaveStyle({
      'background-color': theme.colors.transparent
    })
    expect(screen.getByTestId('circleSubtitle')).toHaveStyle({
      color: theme.colors.spanishGray
    })
  })

  it('should render circle line with spanishGray color when active is false and displayLine is true', () => {
    render(<CircleProgress active={false} displayLine={true}></CircleProgress>)

    expect(screen.getByTestId('circleLine')).toHaveStyle({
      border: '1px solid',
      'border-color': theme.colors.spanishGray
    })
  })

  it('should render circle with icon when an icon is set', () => {
    render(
      <CircleProgress icon={<App data-testid="appIcon" />}></CircleProgress>
    )

    expect(screen.getByTestId('circleIcon')).toContainElement(
      screen.getByTestId('appIcon')
    )
  })

  it('should render subtitle when it is set', () => {
    render(<CircleProgress>Test subtitle</CircleProgress>)

    expect(screen.getByTestId('circleSubtitle')).toHaveTextContent(
      'Test subtitle'
    )
  })
})
