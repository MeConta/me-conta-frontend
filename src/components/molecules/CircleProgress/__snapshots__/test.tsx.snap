// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleProgress/> should render the component on screen 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 9rem;
  max-width: 11rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 7rem;
  height: 7rem;
  background-color: transparent;
  border-radius: 50%;
  -webkit-transition: all 1s;
  transition: all 1s;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 8rem;
  height: 8rem;
  background-color: transparent;
  border-radius: 50%;
  border-color: #A39C9C;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 1s;
  transition: all 1s;
}

.c4 {
  inline-size: 10ch;
  width: inherit;
  color: #A39C9C;
  font-size: 1.5rem;
  text-align: center;
  -webkit-transition: all 1s;
  transition: all 1s;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding-left: 0;
}

.c3 > svg {
  width: 4rem;
  height: 100%;
  color: #A39C9C;
  -webkit-transition: all 1s;
  transition: all 1s;
}

<div
  class="c0"
  data-testid="circleContainer"
>
  <div
    class="c1"
    data-testid="circleLine"
  >
    <div
      class="c2"
      data-testid="circle"
    >
      <div
        class="c3"
        data-testid="circleIcon"
      />
    </div>
  </div>
  <p
    class="c4"
    data-testid="circleSubtitle"
  />
</div>
`;
